app/src/main/java/co/nimblehq/alpine/sample/ui/camera/CameraCaptureViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package co.nimblehq.alpine.sample.ui.camera
+
+import androidx.camera.core.ImageProxy
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.nimblehq.alpine.lib.AlpineLib
+import co.nimblehq.alpine.lib.model.CameraImage
+import co.nimblehq.alpine.lib.model.MrzInfo
+import co.nimblehq.alpine.lib.mrz.MrzProcessorException
+import co.nimblehq.alpine.lib.mrz.MrzProcessorResultListener
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+interface Input {
+
+    fun processImage(
+        cameraImage: CameraImage,
+        imageProxy: ImageProxy
+    )
+
+    fun processImageFile(filePath: String)
+}
+
+interface Output {
+
+    val navigateToNfcScan: SharedFlow<MrzInfo>
+
+    val onProcessImageFailed: SharedFlow<Throwable>
+}
+
+class CameraCaptureViewModel(
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
+) : ViewModel(), Input, Output {
+
+    private val _navigateToNfcScan = MutableSharedFlow<MrzInfo>()
+    override val navigateToNfcScan: SharedFlow<MrzInfo>
+        get() = _navigateToNfcScan
+
+    private val _onProcessImageFailed = MutableSharedFlow<Throwable>()
+    override val onProcessImageFailed: SharedFlow<Throwable>
+        get() = _onProcessImageFailed
+
+    override fun processImage(
+        cameraImage: CameraImage,
+        imageProxy: ImageProxy
+    ) {
+        AlpineLib.processImage(cameraImage, object : MrzProcessorResultListener {
+            override fun onSuccess(mrzInfo: MrzInfo) {
+                viewModelScope.launch(dispatcher) {
+                    imageProxy.close()
+                    _navigateToNfcScan.emit(mrzInfo)
+                }
+            }
+
+            override fun onError(e: MrzProcessorException) {
+                viewModelScope.launch(dispatcher) {
+                    imageProxy.close()
+                    _onProcessImageFailed.emit(e)
```

Comment:
We probably shouldn't emit the error here, otherwise it will spam approx 30 error logs every second 🤔

Reply:
@Wadeewee Yes. `ImageProxy.close()` is used to indicate that we have finished processing that frame, so we need to call it in both `onSuccess` and `onError` before moving on to the next image 😄

Reply:
Ah @ryan-conway You are saying we don't need to handle `error` for live capturing right? just `imageProxy.close()` then the user should consider capturing the image manually instead

Reply:
@Wadeewee Yes exactly 😄 Because it will always call `onError` for every frame until the MRZ is detected 😅